server/services/RuleService.ts

```diff
@@ -0,0 +1,247 @@
+      const { prePackaged } = request.query;
+      const params: GetRulesParams = {
+        prePackaged,
+        body: request.body,
+      };
+      console.log(`Making get rules req: ${params}`);
+      const { callAsCurrentUser: callWithRequest } = this.osDriver.asScoped(request);
+      const getRuleResponse: GetRulesResponse = await callWithRequest(
+        CLIENT_RULE_METHODS.GET_RULES,
+        params
+      );
+
+      return response.custom({
+        statusCode: 200,
+        body: {
+          ok: true,
+          response: getRuleResponse,
+        },
+      });
+    } catch (error: any) {
+      console.error('Security Analytics - RulesService - getRules:', error);
+      return response.custom({
+        statusCode: 200,
+        body: {
+          ok: false,
+          error: error.message,
+        },
+      });
+    }
+  };
+
+  deleteRule = async (
+    _context: RequestHandlerContext,
+    request: OpenSearchDashboardsRequest<DeleteRuleParams, {}>,
+    response: OpenSearchDashboardsResponseFactory
+  ): Promise<IOpenSearchDashboardsResponse<ServerResponse<DeleteRuleResponse> | ResponseError>> => {
+    try {
+      const { callAsCurrentUser: callWithRequest } = this.osDriver.asScoped(request);
+      const deleteRuleResponse: DeleteRuleResponse = await callWithRequest(
+        CLIENT_RULE_METHODS.DELETE_RULE,
+        request.params
+      );
+
+      return response.custom({
+        statusCode: 200,
+        body: {
+          ok: true,
+          response: deleteRuleResponse,
+        },
+      });
+    } catch (error: any) {
+      return response.custom({
+        statusCode: 200,
+        body: {
+          ok: false,
+          error: error.message,
+        },
+      });
+    }
+  };
+
+  /**
+   * Calls backend PUT Rules API.
+   */
+
+  updateRule = async (
+    _context: RequestHandlerContext,
+    request: OpenSearchDashboardsRequest,
+    response: OpenSearchDashboardsResponseFactory
+  ): Promise<IOpenSearchDashboardsResponse<ServerResponse<UpdateRuleResponse> | ResponseError>> => {
+    try {
+      const {
+        id,
+        title,
+        description,
+        detection,
+        status,
+        author,
+        references,
+        tags,
+        log_source,
+        level,
+        false_positives,
+      } = request.body as Rule;
+      const today = moment(moment.now()).format('YYYY/MM/DD');
+      const { ruleId } = request.params as { ruleId: string };
+      const jsonPayload = {
+        id,
+        title,
+        description,
+        status,
+        author,
+        date: today,
+        modified: today,
+        references: references.map((ref) => ref.value),
+        tags: tags.map((tag) => tag.value),
+        logsource: {
+          product: log_source,
+          service: log_source,
+        },
+        level,
+        falsepositives: false_positives.map((falsePos) => falsePos.value),
+        detection: JSON.parse(detection),
+      };
+
+      console.log(jsonPayload);
```

Comment:
Not blocking for this draft, but are these development logs that can be removed, or are we planning to leave this in?

Reply:
Will be removed